config/auth_azure_cli.go

```diff
@@ -50,11 +78,16 @@ func (c AzureCliCredentials) Configure(ctx context.Context, cfg *Config) (func(*
 		return nil, fmt.Errorf("resolve host: %w", err)
 	}
 	logger.Infof(ctx, "Using Azure CLI authentication with AAD tokens")
-	return refreshableVisitor(&ts), nil
+	baseVisitor, err := c.getVisitor(ctx, cfg, ts)
+	if err != nil {
+		return nil, err
+	}
+	return azureVisitor(cfg.AzureResourceID, baseVisitor), nil
```

Comment:
This can be folded into `getVisitor` to make things a bit simpler (i.e. "what is a base visitor?").

Reply:
Sounds good. I did it this way to not have to write azureVisitor twice. I'll adopt your suggestion, but in general I try to only duplicate when things intentionally could/should diverge and not duplicate when things must have identical behavior.